### PR TITLE
LIMS-1113: Shipping service dispatch redirect

### PR DESCRIPTION
**Pull request by MattPrit**
**JIRA ticket**: [LIMS-1113](https://jira.diamond.ac.uk/browse/lims-1113)

**Summary**:
Adds the option to redirect to the shipping service to book shipments when requesting dispatch of appropriate Dewars.

**Changes**:
- Adds feature flag `$use_shipping_service_redirect` which enables/disables this feature
- On requesting Dewar dispatch, create a shipment request in the shipping service and redirect to the appropriate shipment workflow to book a shipment
- Use minimised dispatch request email for shipments booked via the new pathway

**To test**:
- Check that the appropriate dispatch form is displayed for academic, industrial, international users
- Check that the appropriate shipment request is created in the shipping service when requesting dispatch of a domestic academic Dewar
- Check that the minified dispatch-lite email template is used for such dispatch requests; ensure this has an appropriate link to where a Goods Handling operative can view and print the air waybill
- Check that the dispatch button links to the shipping service once a dispatch shipment request has been successfully created.

### CODE REVIEW

File: client/src/js/modules/shipment/views/dispatch.js [hunk 1]
```diff
@@ -112,6 +115,14 @@ define(['marionette', 'views/form',
         
         success: function() {
             app.trigger('shipment:show', this.getOption('dewar').get('SHIPPINGID'))
+            if (app.options.get("shipping_service_app_url") && (this.terms.get('ACCEPTED') == 1)) { // terms.ACCPETED could be undefined, 1, or "1"
```
Comment (gfrn) [inline client/src/js/modules/shipment/views/dispatch.js:118]:
```suggestion
            if (app.options.get("shipping_service_app_url") && (Number(this.terms.get('ACCEPTED')) === 1)) { // terms.ACCPETED could be undefined, 1, or "1"
```

File: api/src/Page/Shipment.php [hunk 2]
```diff
@@ -924,6 +924,60 @@ function _transfer_dewar()
         $this->_output(1);
     }
 
+    function _dispatch_dewar_shipment_request($dewar)
+    {
+        if (is_null($dewar['EXTERNALSHIPPINGIDFROMSYNCHROTRON'])) {
```
Comment (gfrn) [inline api/src/Page/Shipment.php:929]: Consider refactoring this into an early return for readability

**Comment by MattPrit** [comment]
NOTE: Requires the addition of the `externalShippingIdFromSynchrotron` column to the `Dewar` table in ISPyB. Since `_get_dewars` in `Shipment.php` requires this column, and `_get_dewars` is used in multiple endpoints, the shipments table page, the shipment page, and possibly others will not work without this new column.

**Review by gfrn — APPROVED** [review]

**Comment by MattPrit** [comment]
> NOTE: Requires the addition of the `externalShippingIdFromSynchrotron` column to the `Dewar` table in ISPyB. Since `_get_dewars` in `Shipment.php` requires this column, the shipments table page, the shipment page, and possibly others will not work without this new column.

This change has now been made, see https://github.com/DiamondLightSource/ispyb-database/pull/228

**Pull request closed without merging**